extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,26 +1,50 @@
-import {Layout, applyFillContent} from '#core/dom/layout';
+import {Layout} from '#core/dom/layout';
 
-const TAG = 'amp-story-shopping-config';
+import {Services} from '#service';
+
+import {devAssert} from '#utils/log';
+
+import {getRequestService} from '../../amp-story/1.0/amp-story-request-service';
+import {Action} from '../../amp-story/1.0/amp-story-store-service';
 
 export class AmpStoryShoppingConfig extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+    /** @private @const {?./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private @const {!./amp-story-request-service.AmpStoryRequestService} */
+    this.requestService_ = getRequestService(this.win, element);
+  }
+
+  /**
+   * Keys product data to product-ids and adds them to the store service.
+   * @param {!JsonObject} storyConfig
+   * @private
+   */
+  addShoppingStateFromConfig_(storyConfig) {
+    const productIDtoProduct = {};
+
+    for (const item of storyConfig['items']) {
+      productIDtoProduct[item['product-tag-id']] = item;
+    }
 
-    /** @private {string} */
-    this.myText_ = TAG;
+    this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, productIDtoProduct);
 
-    /** @private {?Element} */
-    this.container_ = null;
+    //TODO(#36412): Add call to validate config here.
   }
 
   /** @override */
   buildCallback() {
     super.buildCallback();
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    applyFillContent(this.container_, /* replacedContent */ true);
+    return Services.storyStoreServiceForOrNull(this.win)
+      .then((storeService) => {
+        devAssert(storeService, 'Could not retrieve AmpStoryStoreService');
+        this.storeService_ = storeService;
+        return this.requestService_.loadConfigImpl_(this.element);
```

Comment:
`loadConfigImpl_` will need to be made a public variable, please remove this `_`